source/codegen/metadata_mutation.py

```diff
@@ -85,12 +87,30 @@ def mark_attr_value_param_if_required(function, enums, attribute_enums_by_type,
         enum_type = attribute_value_param["type"]
     if attribute_value_param != None and enum_type in attribute_enums_by_type:
         enum_name = get_attribute_values_enum_name(service_class_prefix, enum_type)
-        if(enum_name in enums):
-            attribute_value_param['enum'] = enum_name
+        if enum_name in enums or f'{enum_name}Mapped' in enums:
```

Comment:
I was going to say that this is a duplicate nested if. But is it? Can you add both mapped and unmapped in the same pass here?

Reply:
Is that covered in existing test cases?